app/validation.go

```diff
@@ -182,6 +182,14 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
 		if app.tokenKeeper.IsTokenFrozen(ctx, msg.Symbol) {
 			return types.ErrTokenFrozen()
 		}
+
```

Comment:
1- here is for fungible token not nin-fungible token.
2- please create a function for check token existence.